src/__tests__/ShoppingList.test.tsx

```diff
@@ -0,0 +1,62 @@
+import { beforeEach, describe, test, expect } from "vitest";
+import { render, screen } from "@testing-library/react";
+import { userEvent } from "@testing-library/user-event";
+import ShoppingList from "@/components/ShoppingList";
+
+describe("ShoppingList", () => {
+  const user = userEvent.setup();
+
+  const addItem = async (itemName: string) => {
+    const input = screen.getByPlaceholderText("Item Name");
+    const addItemButton = screen.getByRole("button", { name: /add item/i });
+    await user.type(input, itemName);
+    await user.click(addItemButton);
+  };
+
+  const removeItem = async (itemName: string) => {
+    const removeItemButton = screen.getByRole("button", {
+      name: new RegExp(`remove ${itemName}`, "i"),
+    });
+    await user.click(removeItemButton);
+  };
+
+  beforeEach(() => {
+    render(<ShoppingList />);
+  });
+
+  describe("Single item opreations", () => {
+    test("should add an item into shopping list", async () => {
```

Comment:
I think it's common to say `it` instead of `test`, but this is probably just down to preference